libs/math/include/math/activation_functions/softmax.hpp

```diff
@@ -99,9 +99,24 @@ void Softmax(ArrayType const &array, ArrayType &ret, typename ArrayType::SizeTyp
   {
     details::Softmax2DImplementation(array, ret, axis);
   }
+  else if ((array.shape().size() == 3) && (ret.shape().size() == 3) && (axis == 0 || axis == 1))
+  {
+    auto tmp_ret = ret.View(0).Copy();
```

Comment:
add a TODO(1320) - Copy()s can be removed when softmax2dimplementation handles IsIterable types